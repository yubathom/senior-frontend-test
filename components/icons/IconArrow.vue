<template>
  <svg
    width="18"
    height="10"
    viewBox="0 0 18 10"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1 0.999999L9 9L17 1"
      stroke="white"
      stroke-linecap="round"
      stroke-linejoin="round"
      :stroke-width="strokeWidth"
    />
  </svg>
</template>
<script>
export default {
  name: 'IconArrow',
  props: {
    strokeWidth: {
      type: Number,
      default: 1
    }
  }
}
</script>
