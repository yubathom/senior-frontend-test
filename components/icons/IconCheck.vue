<template>
  <svg
    width="24"
    height="24"
    viewBox="0 0 24 24"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M5 13L9 17L19 7"
      :stroke="stroke"
      :stroke-width="strokeWidth"
      stroke-linecap="round"
      stroke-linejoin="round"
    />
  </svg>
</template>
<script>
export default {
  name: 'IconCheck',
  props: {
    stroke: {
      type: String,
      default: 'white'
    },
    strokeWidth: {
      type: Number,
      default: 2
    }
  }
}
</script>
