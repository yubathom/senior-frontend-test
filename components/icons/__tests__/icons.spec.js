import { shallowMount } from '@vue/test-utils'
import IconArrow from '../IconArrow'
import IconCheck from '../IconCheck'
import IconPencil from '../IconPencil'
import IconTrash from '../IconTrash'

const icons = [
  {
    title: 'IconArrow',
    component: IconArrow,
    propsData: {
      strokeWidth: 3
    }
  },
  {
    title: 'IconCheck',
    component: IconCheck,
    propsData: {
      stroke: 'pink',
      strokeWidth: 10
    }
  },
  {
    title: 'IconPencil',
    component: IconPencil,
    propsData: {
      stroke: '#fafafa'
    }
  },
  {
    title: 'IconTrash',
    component: IconTrash,
    propsData: {
      stroke: 'purple'
    }
  }
]

const factory = (component, propsData) => {
  return shallowMount(component, { propsData })
}

icons.forEach(({ title, component, propsData }) => {
  describe(title, () => {
    test(`${title} renders properly`, () => {
      const wrapper = factory(component, propsData)
      expect(wrapper.html()).toMatchSnapshot()
    })
  })
})
