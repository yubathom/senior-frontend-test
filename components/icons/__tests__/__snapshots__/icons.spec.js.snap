// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconArrow IconArrow renders properly 1`] = `
<svg width="18" height="10" viewBox="0 0 18 10" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M1 0.999999L9 9L17 1" stroke="white" stroke-linecap="round" stroke-linejoin="round" stroke-width="3"></path>
</svg>
`;

exports[`IconCheck IconCheck renders properly 1`] = `
<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M5 13L9 17L19 7" stroke="pink" stroke-width="10" stroke-linecap="round" stroke-linejoin="round"></path>
</svg>
`;

exports[`IconPencil IconPencil renders properly 1`] = `
<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M15.232 5.23199L18.768 8.76799M16.732 3.73199C17.2009 3.26309 17.8369 2.99966 18.5 2.99966C19.1631 2.99966 19.7991 3.26309 20.268 3.73199C20.7369 4.2009 21.0003 4.83687 21.0003 5.49999C21.0003 6.16312 20.7369 6.79909 20.268 7.26799L6.5 21.036H3V17.464L16.732 3.73199Z" stroke="#fafafa" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path>
</svg>
`;

exports[`IconTrash IconTrash renders properly 1`] = `
<svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
  <path d="M10 11V17M14 11V17M4 7H20M19 7L18.133 19.142C18.0971 19.6466 17.8713 20.1188 17.5011 20.4636C17.1309 20.8083 16.6439 21 16.138 21H7.862C7.35614 21 6.86907 20.8083 6.49889 20.4636C6.1287 20.1188 5.90292 19.6466 5.867 19.142L5 7H19ZM15 7V4C15 3.73478 14.8946 3.48043 14.7071 3.29289C14.5196 3.10536 14.2652 3 14 3H10C9.73478 3 9.48043 3.10536 9.29289 3.29289C9.10536 3.48043 9 3.73478 9 4V7H15Z" stroke="purple" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"></path>
</svg>
`;
